terra/contracts/wormhole/src/contract.rs

```diff
@@ -569,8 +545,45 @@ pub fn handle_set_active<S: Storage, A: Api, Q: Querier>(
 
     config(&mut deps.storage).save(&state)?;
 
+    Ok(HandleResponse::default())
+}
+
+pub fn handle_set_fee<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    fee: Coin,
+) -> StdResult<HandleResponse> {
+    let mut state = config_read(&deps.storage).load()?;
+
+    if deps.api.canonical_address(&env.message.sender)? != state.owner {
+        return ContractError::PermissionDenied.std_err();
+    }
+
+    state.fee = fee;
+
+    config(&mut deps.storage).save(&state)?;
+
+    Ok(HandleResponse::default())
+}
+
+pub fn handle_transfer_fee<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    amount: Coin,
+    recipient: HumanAddr,
+) -> StdResult<HandleResponse> {
+    let state = config_read(&deps.storage).load()?;
+
+    if deps.api.canonical_address(&env.message.sender)? != state.owner {
+        return ContractError::PermissionDenied.std_err();
+    }
+
     Ok(HandleResponse {
-        messages: vec![],
+        messages: vec![CosmosMsg::Bank(BankMsg::Send{
```

Comment:
It is just concern. If you always left $1 balance or feet token is Luna, you don't need to care about this.

In Terra blockchain, we charge tax to all native token transfer action except Luna. 
When the contract try to withdraw fee to other account, the chain will charge a tax from the left contract balance. 

To prevent over-withdraw due to tax, you can pre-compute the tax amount will be charged to this transfer action.

Ex) https://github.com/terraswap/terraswap/blob/72c60c05c43841499f760710a03f864c5ee4db3b/packages/terraswap/src/asset.rs#L32-L56

Reply:
This is a great point, but I think we should keep the current logic for a couple of reasons:

- Fee on the contract is for spam prevention only, so I do not think it will be collected frequently if at all
- This is the admin method called by the contract owner, so I would assume that they know what they're doing

If it was a method a regular user could call this would be a different story and we need to add checks for Tax there.

Reply:
And also yes, I think the fee will be in LUNA :)

Reply:
cool